### ISSUE TITLE
Unable to set permissions for user

### ISSUE THREAD

**Issue by edmundoa** [Graylog2/graylog2-server#1141]
Investigate if setting user permissions through the API works as expected on 1.0.2.

The problem occurs using the swagger UI and doing a PUT to `/users/{username}/permissions` (the user comes from LDAP), with JSON body set to:
```
 {"permissions":["streams:read:554240f3e4b0f343c1fdb8e4","streams:read:554252bce4b0f343c1fdcc2f","dashboards:read:55427ce3e4b0f343c1fdf9ec"]}
```

The response from the server is:
```
{ "type": "ApiError",   "message": "Can not deserialize instance of java.util.ArrayList out of FIELD_NAME token\n at [Source: org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream@4622055f; line: 2, column: 3]" }
```

I tried to quickly reproduce the issue on master, but it worked as expected.

**Comment by georgefrank** [comment]
I reported this on freenode IRC. It also happens for me when I make the request via the POSTMAN Chrome extension.

**Comment by joschi** [comment]
Looks like this is an unfortunate mishap with the definition of this request in Graylog 1.0.x.

The following request body (or curl request) will work for Graylog 1.0.x:
```json
["streams:read:554240f3e4b0f343c1fdb8e4","streams:read:554252bce4b0f343c1fdcc2f","dashboards:read:55427ce3e4b0f343c1fdf9ec"]
```
```
curl -i -u admin:$PASSWORD \
  -H 'Content-Type: application/json' \
  -X PUT \
  -d '["streams:read:554240f3e4b0f343c1fdb8e4","streams:read:554252bce4b0f343c1fdcc2f","dashboards:read:55427ce3e4b0f343c1fdf9ec"]' \
  http://localhost:12900/users/$USER/permissions
```

The issue has already been fixed in Graylog 1.1.0-SNAPSHOT but most likely won't be back-ported into Graylog 1.0.3-SNAPSHOT.

**Comment by georgefrank** [comment]
Great! Thanks for the workaround!

**Comment by edmundoa** [comment]
As it seems to work and this is fixed on Graylog 1.1.0-SNAPSHOT, I'm closing the issue. Please reopen if needed!
